boxsdk/object/file.py

```diff
@@ -79,34 +79,66 @@ def _get_accelerator_upload_url_for_update(self):
         return self._get_accelerator_upload_url(file_id=self._object_id)
 
     @api_call
-    def content(self):
+    def content(self, file_version=None, byte_range=None):
         """
         Get the content of a file on Box.
 
+        :param file_version:
+            The specific version of the file to retrieve the contents of.
+        :type file_version:
+            :class:`FileVersion` or None
+        :param byte_range:
+            A tuple of inclusive byte offsets to download, e.g. (100, 199) to download the second 100 bytes of a file
+        :type byte_range:
+            (`int`, `int`)
         :returns:
             File content as bytes.
         :rtype:
             `bytes`
         """
         url = self.get_url('content')
-        box_response = self._session.get(url, expect_json_response=False)
+        params = {'version': file_version.object_id} if file_version is not None else None
+        headers = {'Range': 'bytes={0}-{1}'.format(*byte_range)} if byte_range is not None else None
+        box_response = self._session.get(url, expect_json_response=False, params=params, headers=headers)
         return box_response.content
 
     @api_call
-    def download_to(self, writeable_stream):
+    def download_to(self, writeable_stream, file_version=None, byte_range=None):
         """
         Download the file; write it to the given stream.
 
         :param writeable_stream:
             A file-like object where bytes can be written into.
         :type writeable_stream:
             `file`
+        :param file_version:
+            The specific version of the file to retrieve the contents of.
+        :type file_version:
+            :class:`FileVersion` or None
+        :param byte_range:
+            A tuple of inclusive byte offsets to download, e.g. (100, 199) to download the second 100 bytes of a file
+        :type byte_range:
+            (`int`, `int`)
         """
         url = self.get_url('content')
-        box_response = self._session.get(url, expect_json_response=False, stream=True)
+        params = {'version': file_version.object_id} if file_version is not None else None
+        headers = {'Range': 'bytes={0}-{1}'.format(*byte_range)} if byte_range is not None else None
```

Comment:
We shouldn't to duplicate this logic, we should have a class for it (as I described above) or at least a helper method in this class.